tests/sql_injection_test.py

```diff
@@ -0,0 +1,240 @@
+"""Project: New Arrivals Chi.
+
+File name: sql_injection_test.py
+Associated Files: new_arrivals_chi/app/main.py,
+                  new_arrivals_chi/app/logger_config.py, tests/db_test.py.
+
+
+This test suite performs testing on the database to ensure that input handling,
+particularly regarding SQL injections, is secure and prevents unauthorized
+database access or manipulation.
+
+
+Methods:
+    - test_safe_injections: Test function for testing safe input handling or
+                            prevention of SQL injections.
+    - test_unsafe_injections_return_all_rows: Test function for an unsafe input
+                                              attempting to return all rows from
+                                              the database.
+    - test_unsafe_injections_alter_table: Test function for an unsafe input
+                                          attempting to alter the database table
+                                          structure.
+    - test_line_comments_injection: Test function for testing prevention of line
+                                    comments in SQL queries.
+    - test_union_injection: Test function for testing prevention of Union-based
+                            SQL injections.
+    - test_unsafe_error_based_injection: Test function for testing error-based
+                                         SQL injections that find column names.
+    - test_unsafe_boolean_based_blind_sql_injection: Test function for simulat-
+                                                    ing an SQL injection attack
+                                                    using an IF statement.
+
+
+
+
+Last updated:
+@Author: Xiomara Salazar @xiomara0
+@Date: 05/09/2024
+
+Creation:
+@Author: Xiomara Salazar @xiomara0
+@Date: 05/09/2024
+"""
+
+from sqlalchemy import select
+from .db_test import create_fake_user, create_fake_organization
+from .setup_fake_db import main
+from new_arrivals_chi.app.database import User
+
+
+# creating fake users for queries
+test_org = create_fake_organization()
+test_user = create_fake_user(test_org)
+
+
+def database_query(setup_logger, input_value):
+    """Executing a database query and check the number of rows returned.
+
+    Executes a database query and checks the number of returned rows based on
+    an input value that filters the email column.
+
+    Args:
+        setup_logger: Setup logger.
+        input_value: Input value for the database query that filters on email
+    """
+    test_app, test_db = main()
+
+    logger = setup_logger("test_injections")
+
+    try:
+        with test_app.app_context():
+            logger.info("Starting injection test")
+
+            session = test_db.session
+            logger.info("Attempting to retrieve user with test input")
+            filtered_users_select = select(User).where(
+                User.email == input_value
+            )
```

Comment:
This still risks SQL injection. We should use parametrized queries here with bind parameters. Here's a free one for `database_query`.

```
from sqlalchemy import select, text

filtered_users_select = select(User).where(User.email == text(":email"))
all_users_select = select(User)

all_users = session.execute(all_users_select).scalars().all()
filtered_users = session.execute(filtered_users_select, {'email': input_value}).scalars().all()

```

Generally you don't want to interpolate information directly to (concatenate to) the SQL query since that's where an injection would occur.

Reply:
Updated!